source/node_test.go

```diff
@@ -321,6 +341,7 @@ func testNodeSourceEndpoints(t *testing.T) {
 			"node with nil Lables returns valid endpoint",
 			"",
 			"",
+			labels.Everything(),
```

Comment:
Can you please add a test case for when the the label is actually filters some nodes?